presto-hive/src/main/java/io/prestosql/plugin/hive/ViewReaderUtil.java

```diff
@@ -0,0 +1,229 @@
+        {
+            checkCondition(viewData.startsWith(VIEW_PREFIX), HIVE_INVALID_VIEW_DATA, "View data missing prefix: %s", viewData);
+            checkCondition(viewData.endsWith(VIEW_SUFFIX), HIVE_INVALID_VIEW_DATA, "View data missing suffix: %s", viewData);
+            viewData = viewData.substring(VIEW_PREFIX.length());
+            viewData = viewData.substring(0, viewData.length() - VIEW_SUFFIX.length());
+            byte[] bytes = Base64.getDecoder().decode(viewData);
+            return VIEW_CODEC.fromJson(bytes);
+        }
+    }
+
+    /**
+     * Class to decode Hive view definitions
+     */
+    public static class HiveViewDecoder
+            implements ViewReader
+    {
+        private final HiveMetastoreClient mscClient;
+        private final TypeManager typeManager;
+
+        public HiveViewDecoder(HiveMetastoreClient mscClient, TypeManager typemanager)
+        {
+            this.mscClient = mscClient;
+            this.typeManager = requireNonNull(typemanager, "metadata is null");
+        }
+
+        @Override
+        public ConnectorViewDefinition decodeViewData(String viewSql, Table table, CatalogName catalogName)
+        {
+            try {
+                HiveToRelConverter hiveToRelConverter = HiveToRelConverter.create(mscClient);
+                RelNode rel = hiveToRelConverter.convertView(table.getDatabaseName(), table.getTableName());
+                RelToPrestoConverter rel2Presto = new RelToPrestoConverter();
+                String prestoSql = rel2Presto.convert(rel);
+                RelDataType rowType = rel.getRowType();
+                List<ConnectorViewDefinition.ViewColumn> columns = new ArrayList<>(rowType.getFieldCount());
+                for (RelDataTypeField field : rowType.getFieldList()) {
+                    log.debug("Adding column %s, %s, %s, %s", field.getName(),
+                            field.getType().getFullTypeString(),
+                            field.getType().getSqlTypeName(),
+                            field);
+
+                    Type type = typeManager.fromSqlType(getTypeString(field.getType()));
+                    ConnectorViewDefinition.ViewColumn column = new ConnectorViewDefinition.ViewColumn(field.getName(), type.getTypeId());
+                    columns.add(column);
+                }
+                return new ConnectorViewDefinition(prestoSql,
+                        Optional.of(catalogName.toString()),
+                        Optional.of(table.getDatabaseName()),
+                        columns,
+                        Optional.ofNullable(table.getParameters().get(TABLE_COMMENT)),
+                        Optional.empty(),
+                        true);
+            }
+            catch (RuntimeException e) {
+                throw new PrestoException(HIVE_UNKNOWN_ERROR,
+                        format("Error decoding view definition for %s.%s: %s. Please report this to ask_dali@linkedin.com",
```

Comment:
I think you want to remove this second part :)

Is it useful to duplicate the error message here? It's available in the stack trace. Let's also add quotes around the table name, and use the right error code.
```java
throw new PrestoException(HIVE_VIEW_TRANSLATION_ERROR, format("Failed to translate Hive view '%s': %s', table.getSchemaTableName(), e.getMessage()), e);